.github/workflows/pr-checker.yml

```diff
@@ -0,0 +1,99 @@
+name: pullrequest-build-and-scan
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - ci_dev
+      - ci_prod
+    paths-ignore:
+      - '**.md'
+jobs:
+  LINUX-build-and-scan:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set-workflow-initiator
+        run: echo "Initiated by - ${GITHUB_ACTOR}"
+      - name: Set-branch-name-for-pr
+        if: ${{ github.event_name == 'pull_request' }}
+        run: echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF} | tr / _)" >> $GITHUB_ENV
+      - name: Set-Env
+        run: echo "ENV=dev" >> $GITHUB_ENV
+      - name: Set-ACR-Registry
+        run: echo "ACR_REGISTRY=containerinsightsprod.azurecr.io" >> $GITHUB_ENV
+      - name: Set-ACR-Repository
+        run: echo "ACR_REPOSITORY=/public/azuremonitor/containerinsights/cidev" >> $GITHUB_ENV
+      - name: Set-image-tag-name
+        run: echo "IMAGE_TAG_NAME=cidev" >> $GITHUB_ENV
+      - name: Set-image-tag-suffix
+        run: echo "IMAGE_TAG_DATE=$(date +%m-%d-%Y)" >> $GITHUB_ENV
+      - name: Set-commit-sha
+        run: echo "COMMIT_SHA=${GITHUB_SHA::8}" >> $GITHUB_ENV
+      - name: Set-image-tag
+        run: echo "IMAGETAG=${ACR_REGISTRY}${ACR_REPOSITORY}:${IMAGE_TAG_NAME}-${BRANCH_NAME}-${IMAGE_TAG_DATE}-${COMMIT_SHA}" >> $GITHUB_ENV
+      - name: Set-image-telemetry-tag
+        run: echo "IMAGETAG_TELEMETRY=${IMAGE_TAG_NAME}-${BRANCH_NAME}-${IMAGE_TAG_DATE}-${COMMIT_SHA}" >> $GITHUB_ENV
+      - name: Set-Helm-OCI-Experimental-feature
+        run: echo "HELM_EXPERIMENTAL_OCI=1" >> $GITHUB_ENV
+      - name: Set-Helm-chart-version
+        run: echo "HELM_CHART_VERSION=0.0.1" >> $GITHUB_ENV
+      - name: Set-Helm-tag
+        run: echo "HELMTAG=${ACR_REGISTRY}${ACR_REPOSITORY}:${IMAGE_TAG_NAME}-chart-${BRANCH_NAME}-${HELM_CHART_VERSION}-${IMAGE_TAG_DATE}-${COMMIT_SHA}" >> $GITHUB_ENV
```

Comment:
Do we need the Helm related steps for docker-provider?

Reply:
i plan to enable this for the future.